release/ray_release/test_automation/ci_state_machine.py

```diff
@@ -46,24 +63,39 @@ def _create_github_issue(self) -> None:
             "triage",
             self.test.get_oncall(),
         ]
+        recent_failures = [
+            result for result in self.test_results if result.is_failing()
+        ][:3]
+        body = (
+            f"CI test **{self.test.get_name()}** is {self.test.get_state().value}. "
+            "Recent failures: \n"
+        )
+        for failure in recent_failures:
+            body += f"\t- {failure.url}\n"
+        body += "\nManaged by OSS Test Policy"
         issue_number = self.ray_repo.create_issue(
-            title=f"CI test {self.test.get_name()} failed",
-            body=(
-                f"CI test **{self.test.get_name()}** failed. "
-                f"See {self.test_results[0].url} for more details.\n\n"
-                f"Managed by OSS Test Policy"
-            ),
+            title=f"CI test {self.test.get_name()} is {self.test.get_state().value}",
+            body=body,
             labels=labels,
             assignee=DEFAULT_ISSUE_OWNER,
         ).number
         self.test[Test.KEY_GITHUB_ISSUE_NUMBER] = issue_number
 
     def _consistently_failing_to_jailed(self) -> bool:
-        # CI tests go to flaky first after failed instead of jailed
         return False
 
     def _passing_to_flaky(self) -> bool:
-        # TODO(can): if flaky percentage is too high, to be implemented
+        flake = 0
+        for i in range(1, len(self.test_results) - 1):
+            if (
+                self.test_results[i].is_failing()
+                and self.test_results[i + 1].is_passing()
+                and self.test_results[i - 1].is_passing()
```

Comment:
this means `PPFFPP` will count as no flaky? does not feel quite right to me.

Reply:
yes, there can be more patterns, let expand it more as we go; for now, i review go/flaky and think that this pattern is good enough to catch what we want